DependencyInjection/Compiler/SerializerConfigurationPass.php

```diff
@@ -26,6 +27,10 @@
 {
     public function process(ContainerBuilder $container)
     {
+        if (!class_exists(XmlEncoder::class)) {
+            $container->removeDefinition('fos_rest.decoder.xml');
```

Comment:
This does not solve the problem you described. The service name is used in https://github.com/FriendsOfSymfony/FOSRestBundle/blob/5e38509c14798579ba5655846cc9c4876f0b40e7/DependencyInjection/Configuration.php#L287 and the service is tried to be fetched from the container

So insead the `Class 'Symfony\Component\Serializer\Encoder\XmlEncoder' not found`, this causes a service does not exist error. So the better solution would be to remove the decoder registration in the config I linked above when the serializer is not installed.

Reply:
Humm right, nice catch. 

https://github.com/FriendsOfSymfony/FOSRestBundle/blob/5e38509c14798579ba5655846cc9c4876f0b40e7/DependencyInjection/FOSRestExtension.php#L164

So checking if the "id" is a service here would work and help preventing bad configuration, right?

Reply:
@Tobion  ?

Reply:
Service existence checks should not be done in the extension as it then depends on the order the extensions are registered. As I said, you could check if the class exists and adjust the config default accordingly.